tchannel/tornado/connection.py

```diff
@@ -358,8 +365,15 @@ def _write(self, message):
             payload=payload
         )
         body = frame.frame_rw.write(f, BytesIO()).getvalue()
+        return self._write_queue.put(body)
 
-        return self.connection.write(body)
+    @tornado.gen.coroutine
+    def _drain_write_queue(self):
+        """Ensures ordering of our writes."""
+
+        while not self.closed:
+            body = yield self._write_queue.get()
```

Comment:
problem with this `get` is it will have timeout. It blocks until timeout. So we need to catch the timeout .